File: server/models/order.js
/**
 * 订单模型 → orders 表
 *
 * 订单状态常量（status 字段）：
 * 0 待支付 | 1 已支付（待发货） | 2 配送中 | 3 已完成 | 4 已取消
 */
const { DataTypes } = require('sequelize')
const sequelize = require('../config/database')

const Order = sequelize.define('Order', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  order_no: {
    type: DataTypes.STRING(32),
    allowNull: false,
    unique: true,
    comment: '订单号（时间戳 + 随机数生成）',
  },
  user_id: { type: DataTypes.INTEGER, allowNull: false, comment: '下单用户 id' },
  address_id: { type: DataTypes.INTEGER, allowNull: false, comment: '收货地址 id' },
  // 以下为地址快照：地址被删除/修改后订单仍保留下单时的信息
  receiver: { type: DataTypes.STRING(50), allowNull: false, comment: '收货人（快照）' },
  receiver_phone: { type: DataTypes.STRING(20), allowNull: false, comment: '收货电话（快照）' },
  receiver_address: { type: DataTypes.STRING(255), allowNull: false, comment: '完整收货地址（快照）' },
  total_amount: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
    comment: '商品总额（元）',
  },
  freight: {
    type: DataTypes.DECIMAL(10, 2),
    defaultValue: 0,
    comment: '运费（元）',
  },
  pay_amount: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
    comment: '实付金额 = 总额 + 运费（元）',
  },
  status: {
    type: DataTypes.TINYINT,
    defaultValue: 0,
    comment: '0 待支付 1 已支付 2 配送中 3 已完成 4 已取消',
  },
  remark: { type: DataTypes.STRING(255), defaultValue: '', comment: '买家备注' },
  paid_at: { type: DataTypes.DATE, comment: '支付时间' },
}, {
  tableName: 'orders',
})

module.exports = Order

File: springboot/src/main/java/com/freshmall/entity/Order.java
package com.freshmall.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 订单表
 * 状态：0 待支付 | 1 已支付（待发货） | 2 配送中 | 3 已完成 | 4 已取消
 */
@Data
@Entity
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "order_no", nullable = false, unique = true, length = 32)
    private String order_no;

    @Column(name = "user_id", nullable = false)
    private Integer user_id;

    @Column(name = "address_id", nullable = false)
    private Integer address_id;

    /** 以下为地址快照 */
    @Column(nullable = false, length = 50)
    private String receiver;

    @Column(name = "receiver_phone", nullable = false, length = 20)
    private String receiver_phone;

    @Column(name = "receiver_address", nullable = false, length = 255)
    private String receiver_address;

    @Column(name = "total_amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal total_amount;

    @Column(precision = 10, scale = 2)
    private BigDecimal freight = BigDecimal.ZERO;

    @Column(name = "pay_amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal pay_amount;

    /** 0 待支付 1 已支付 2 配送中 3 已完成 4 已取消 */
    private Integer status = 0;

    @Column(length = 255)
    private String remark = "";

    @Column(name = "paid_at")
    private LocalDateTime paid_at;

    @Column(name = "created_at", updatable = false)
    @CreationTimestamp
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
